lib/AST/ASTWalker.cpp

```diff
@@ -430,11 +430,13 @@ class Traversal : public ASTVisitor<Traversal, Expr*, Stmt*,
   }
 
   bool visitMacroExpansionDecl(MacroExpansionDecl *MED) {
-    if (MED->getArgs() && doIt(MED->getArgs()))
```

Comment:
I wonder if we need a customization point here, because it seems like some things really want the see the original source, other things want to see the rewritten source, and some will want to see both.